### PR TITLE
Signin-SignUp-Auth

### PR DESCRIPTION
**Pull request by ark2002**
- Added user account management

Preview:

https://deploy-preview-5--astromarts.netlify.app/

deploy

### CODE REVIEW

**Review by tushargoyal812 — COMMENTED** [review]
liked how you structured files.
variables names are good.
clean and readable code.

File: src/frontend/screens/SignInScreen/SignInScreen.js [hunk 1]
```diff
@@ -1,8 +1,71 @@
-import React from "react";
+import React, { useState } from "react";
```
Comment (VishakAmin) [inline src/frontend/screens/SignInScreen/SignInScreen.js:1]: Separate the local imports with module imports will increase the readability

Example:
import { Link, useNavigate } from "react-router-dom";

import "./SignInScreen.css";
import { useAuth } from "../../contexts";

**Review by VishakAmin — COMMENTED** [review]
Looks good added a comment please have a look.

File: src/frontend/screens/SignInScreen/SignInScreen.js [hunk 2]
```diff
@@ -1,8 +1,71 @@
...
+            navigate("/");
+        }
+    };
+
     return (
-        <div>SignInScreen</div>
+        <div className="login__container flex--column">
+            <form className=" login__form flex--column" onSubmit={(e) => {
+                e.preventDefault();
+                signInHandler(user, setAuth, navigate);
```
Comment (MarkVed17) [inline src/frontend/screens/SignInScreen/SignInScreen.js]: your `signInHandler` just takes `user` as a parameter. `setAuth` & `navigate` parameters aren't required.

File: src/frontend/screens/SignUpScreen/SignUpScreen.js [hunk 3]
```diff
@@ -1,8 +1,58 @@
...
+            navigate("/");
+        }
+    };
+
     return (
-        <div>SignUpScreen</div>
+        <div className="signup__container flex--column">
+            <form className="signup__form flex--column" onSubmit={(e) => {
+                e.preventDefault();
+                signUpHandler(user, setAuth, navigate);
```
Comment (MarkVed17) [inline src/frontend/screens/SignUpScreen/SignUpScreen.js]: Here as well.
your `signUpHandler` just takes `user` as a parameter. `setAuth` & `navigate` parameters aren't required.

File: src/frontend/screens/SignInScreen/SignInScreen.js (hunk 2 above)
Comment (ark2002) [reply src/frontend/screens/SignInScreen/SignInScreen.js]: yup would look into it

File: src/frontend/services/auth-services/signInService.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import axios from "axios";
+
+export const signInService = async (user) => {
+    try {
+        const { data } = await axios.post("/api/auth/login", {
+            email: user.email,
+            password: user.password,
+        });
+        console.log(data);
```
Comment (MarkVed17) [inline src/frontend/services/auth-services/signInService.js:9]: you may remove this.

File: src/frontend/screens/SignInScreen/SignInScreen.js (hunk 1 above)
Comment (ark2002) [reply src/frontend/screens/SignInScreen/SignInScreen.js:1]: yup would take care of it

**Review by MarkVed17 — APPROVED** [review]
Auth functions smoothly. I liked that you redirected all the other routes to Home. Nice addition.

**Pull request merged**